### ISSUE TITLE
Wrong SQL generated for connectByRoot(minus(one()))

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#8681]
In Oracle `CONNECT_BY_ROOT` is an unary operator with a given precedence. This for instance means that `CONNECT_BY_ROOT 1 * 1` actually means `(CONNECT_BY_ROOT 1) * 1`, which can be of importance in some situations.

Oracle also cannot parse `CONNECT_BY_ROOT -1`. It has to be written as `CONNECT_BY_ROOT (-1)`. jOOQ's behavior is correct in this respect, however when SQL is generated for `connectByRoot(minus(one()))` the result is `CONNECT_BY_ROOT -1`, which is wrong as just discussed.

**Comment by lukaseder** [comment]
Interesting. In that case, definitely, `CONNECT_BY_ROOT` has to be parsed as a unary operator, with the correct operator precedence.

**Comment by lukaseder** [comment]
The test introduced here now passes because of a change in https://github.com/jOOQ/jOOQ/issues/13581

```java
    @Test
    public void testSelectConnectByRootOperatorPrecedence() {
        assertQueryEquals(dsl.select(one).where(connectByRoot(minus(one)).eq(minus(one))).connectBy(level().lt(two)), "select 1 from dual where connect_by_root (-1) = -1 connect by level < 2");
    }
```

But the test didn't test the claims made in this issue. A new formatting test seems to be required:

```java
    @Test
    public void testUnaryOpsConnectByRootOperatorPrecedence() {
        assertFormattingEquals("connect_by_root (-1)", connectByRoot(minus(c1)));
    }
```

That test still fails with:

```
org.junit.ComparisonFailure: Unformatted comparison expected:<connect_by_root [(-1)]> but was:<connect_by_root [-1]>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.jooq.test.AbstractTest.assertEquals(AbstractTest.java:202)
	at org.jooq.test.formatting.AbstractFormattingTest.assertFormattingEquals0(AbstractFormattingTest.java:183)
	at org.jooq.test.formatting.AbstractFormattingTest.assertFormattingEquals(AbstractFormattingTest.java:171)
	at org.jooq.test.formatting.FormattingExpressionTest.testUnaryOpsConnectByRootOperatorPrecedence(FormattingExpressionTest.java:231)
```

**Comment by lukaseder** [comment]
Hmm, we could add the parentheses only if the unary operator's argument isn't already "parenthesised". E.g. for expressions like `(1 + 2)` or `TAB.COL`, we don't need extra parentheses like `CONNECT_BY_ROOT ((1 + 2))` or `CONNECT_BY_ROOT (TAB.COL)`. This is precisely what https://github.com/jOOQ/jOOQ/issues/13581 is about, to avoid unnecessary parentheses.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0. As this fix depends on all the removals of unnecessary parentheses around #13581 and other similar changes in 3.17.0, I cannot backport it in isolation.

Given its low priority, the lack of backport is acceptable.
